nautobot/core/ui/object_detail.py

```diff
@@ -484,8 +508,23 @@ def get_extra_context(self, context):
         for listing and adding objects. It also handles field inclusion/exclusion and
         displays the appropriate table title if provided.
         """
-        body_content_table = context[self.table_key]
+        body_content_table_class = self.table_class
+        body_content_table_model = body_content_table_class.Meta.model
         request = context["request"]
+        instance = get_obj_from_context(context)
+        object_manager = None  # This is to make pylint happy
+        if self.table_attribute:
+            object_manager = getattr(instance, self.table_attribute)
+        if self.table_filter:
+            object_manager = body_content_table_model.objects.filter(**{self.table_filter: instance})
+        body_content_table_queryset = (
+            object_manager.restrict(request.user, "view")
+            .select_related(*self.select_related_fields or [])
+            .prefetch_related(*self.prefetch_related_fields or [])
```

Comment:
Per the docs, `select_related()` with no args actually does a select related on *all* foreign keys, which is probably not what we want here! So this might need to be:

```suggestion
        if self.select_related_fields:
            body_content_table_queryset = body_content_table_queryset.select_related(*self.select_related_fields)
        if self.prefetch_related_fields:
            body_content_table_queryset = body_content_table_queryset.prefetch_related(*self.prefetch_related_fields)
```